### ISSUE TITLE
NUM_FPN 这个参数什么含义

### ISSUE THREAD

**Issue by Arthur-Shi** [DetectionTeamUCAS/NAS_FPN_Tensorflow#10]
代码中：
def resnet_base(img_batch, scope_name, is_training=True):
    if scope_name.endswith('b'):
        get_resnet_fn = get_resnet_v1_b_base
    elif scope_name.endswith('d'):
        get_resnet_fn = get_resnet_v1_d_base
    else:
        raise ValueError("scope Name erro....")

    _, feature_dict = get_resnet_fn(input_x=img_batch, scope=scope_name,
                                    bottleneck_nums=BottleNeck_NUM_DICT[scope_name],
                                    base_channels=BASE_CHANNELS_DICT[scope_name],
                                    is_training=is_training, freeze_norm=True,
                                    freeze=cfgs.FREEZE_BLOCKS)

    pyramid_dict = {}
    with tf.variable_scope('build_pyramid'):
        with slim.arg_scope([slim.conv2d], weights_regularizer=slim.l2_regularizer(cfgs.WEIGHT_DECAY),
                            activation_fn=None, normalizer_fn=None):

            P5 = slim.conv2d(feature_dict['C5'],
                             num_outputs=cfgs.FPN_CHANNEL,
                             kernel_size=[1, 1],
                             stride=1, scope='build_P5')

            pyramid_dict['P5'] = P5

            for level in range(4, 1, -1):  # build [P4, P3, P2]

                pyramid_dict['P%d' % level] = fusion_two_layer(C_i=feature_dict["C%d" % level],
                                                               P_j=pyramid_dict["P%d" % (level + 1)],
                                                               scope='build_P%d' % level)
            for level in range(5, 1, -1):
                pyramid_dict['P%d' % level] = slim.conv2d(pyramid_dict['P%d' % level],
                                                          num_outputs=cfgs.FPN_CHANNEL, kernel_size=[3, 3],
                                                          padding="SAME", stride=1, scope="fuse_P%d" % level,
                                                          activation_fn=tf.nn.relu if cfgs.USE_RELU else None)
            if "P6" in cfgs.LEVLES:
                P6 = slim.avg_pool2d(P5, kernel_size=[1, 1], stride=2, scope='build_P6')
                pyramid_dict['P6'] = P6

    # for level in range(5, 1, -1):
    #     add_heatmap(pyramid_dict['P%d' % level], name='Layer%d/P%d_heat' % (level, level))

    for i in range(cfgs.NUM_FPN):
        pyramid_dict = fpn(pyramid_dict, i)

    for i in range(cfgs.NUM_NAS_FPN):
        pyramid_dict = nas_fpn(pyramid_dict, i)

您在cfgs.py这个文件默认值为0, 为什么在已经搭建好的fpn网络之后，还要执行这个fpn函数尼？
我的理解是nas_fpn这个函数是在搜索fpn,可之前的是什么意义？以及这个参数？
感谢您！

**Comment by yangxue0827** [comment]
只是看原始fpn堆叠有什么效果而已 @Arthur-Shi

**Comment by Arthur-Shi** [comment]
所以在训练中，还是设置为0咯？

我做了cfgs.NUM_NAS_FPN为 384 和256的训练，发现384的效果显著提升。
该数值还能设置的更大吗？

以及在论文中 有关于DropBlock的实验，不知道您代码中有做对应的实现吗？

**Comment by yangxue0827** [comment]
没有做那么详细的实验，你可以自己试一下 @Arthur-Shi

**Comment by Arthur-Shi** [comment]
谢谢，麻烦您了
